internal/auth0/organization/data_source.go

```diff
@@ -0,0 +1,237 @@
+		Type:         schema.TypeString,
+		Optional:     true,
+		Description:  "The ID of the organization. If not provided, `name` must be set.",
+		AtLeastOneOf: []string{"organization_id", "name"},
+	}
+
+	internalSchema.SetExistingAttributesAsOptional(dataSourceSchema, "name")
+	dataSourceSchema["name"].Description = "The name of the organization. If not provided, `organization_id` must be set."
+	dataSourceSchema["name"].AtLeastOneOf = []string{"organization_id", "name"}
+
+	dataSourceSchema["connections"] = &schema.Schema{
+		Type:     schema.TypeSet,
+		Computed: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"connection_id": {
+					Type:        schema.TypeString,
+					Computed:    true,
+					Description: "The ID of the enabled connection on the organization.",
+				},
+				"assign_membership_on_login": {
+					Type:     schema.TypeBool,
+					Computed: true,
+					Description: "When `true`, all users that log in with this connection will be " +
+						"automatically granted membership in the organization. When `false`, users must be " +
+						"granted membership in the organization before logging in with this connection.",
+				},
+			},
+		},
+	}
+
+	dataSourceSchema["members"] = &schema.Schema{
+		Type:     schema.TypeSet,
+		Computed: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"user_id": {
+					Type:        schema.TypeString,
+					Computed:    true,
+					Description: "The ID of the organization member.",
+				},
+				"name": {
+					Type:        schema.TypeString,
+					Computed:    true,
+					Description: "The name of the organization member.",
+				},
+				"email": {
+					Type:        schema.TypeString,
+					Computed:    true,
+					Description: "The email of the organization member.",
+				},
+			},
+		},
+	}
+
+	return dataSourceSchema
+}
+
+func readOrganizationForDataSource(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api := meta.(*management.Management)
+	var foundOrganization *management.Organization
+	var err error
+
+	organizationID := data.Get("organization_id").(string)
+	if organizationID != "" {
+		foundOrganization, err = api.Organization.Read(organizationID)
+		if err != nil {
+			if mErr, ok := err.(management.Error); ok && mErr.Status() == http.StatusNotFound {
+				data.SetId("")
+				return nil
+			}
+			return diag.FromErr(err)
+		}
+	} else {
+		name := data.Get("name").(string)
+		page := 0
+
+	outerLoop:
+		for {
+			organizations, err := api.Organization.List(management.Page(page))
+			if err != nil {
+				return diag.FromErr(err)
+			}
+
+			for _, organization := range organizations.Organizations {
+				if organization.GetName() == name {
+					foundOrganization = organization
+					break outerLoop
+				}
+			}
+
+			if !organizations.HasNext() {
+				break
+			}
+
+			page++
+		}
+
+		if foundOrganization == nil {
+			return diag.Errorf("No organization found with \"name\" = %q", name)
+		}
```

Comment:
I think that it is ok to provide the ability to lookup by name. But given that some tenants have thousands of organizations, might we suggest to the end user to lookup by ID for most performant results? I can just see this loop taking a very long time and users not understanding why.